tokio-io/src/split.rs

```diff
@@ -23,6 +23,17 @@ pub fn split<T: AsyncRead + AsyncWrite>(t: T) -> (ReadHalf<T>, WriteHalf<T>) {
     (ReadHalf { handle: a }, WriteHalf { handle: b })
 }
 
+/// Reunite a previously split resource.
+///
+/// If the given `ReadHalf` and `WriteHalf` do not originate from the
+/// same `AsyncRead::split` operation they will be returned as is in
+/// `Result::Err`.
+pub fn unsplit<T>(r: ReadHalf<T>, w: WriteHalf<T>) -> Result<T, (ReadHalf<T>, WriteHalf<T>)> {
+    r.handle.reunite(w.handle).map_err(|e| {
+        (ReadHalf { handle: e.0 }, WriteHalf { handle: e.1 })
```

Comment:
I think it may be a good idea to wrap this in an error type possibly? It seems the `BiLock` implementation does that. https://docs.rs/futures/0.1.25/futures/stream/struct.ReuniteError.html